assets/js/src/settings/store/normalize_settings.ts

```diff
@@ -0,0 +1,219 @@
+import _ from 'lodash';
+import { t } from 'common/functions';
+import { Settings } from './types';
+
+export default function normalizeSettings(data: any): Settings {
+  const text = asString('');
+  const disabledCheckbox = asBoolean('1', '0', '0');
+  const enabledCheckbox = asBoolean('1', '0', '1');
+  const disabledRadio = asBoolean('1', '', '');
+  const enabledRadio = asBoolean('1', '', '1');
```

Comment:
The aim of this task is to "normalize" data values. This, however, creates many different options how to save boolean. Maybe I don't understand the reasons but I don't think this is a good idea. Instead, I think in our `Settings` type boolean should always be boolean. Not that false is `0` for some fields, `''` for others, and never actually `false`. Don't you think?

Reply:
I totally agree with you that we should have a unique way to represent things, but right now we don't. Changing that now will change what we send to the API and may break the plugin behaviour somewhere. I am trying to keep the save values (even if they are wrong) to not have to refactor the PHP code (which will not be easy).